src/t_zset.c

```diff
@@ -3764,16 +3770,31 @@ void zrankGenericCommand(client *c, int reverse) {
     robj *ele = c->argv[2];
     robj *zobj;
     long rank;
+    int opt_withscores = 0;
+    double score;
 
     if ((zobj = lookupKeyReadOrReply(c,key,shared.null[c->resp])) == NULL ||
         checkType(c,zobj,OBJ_ZSET)) return;
 
     serverAssertWithInfo(c,ele,sdsEncodedObject(ele));
-    rank = zsetRank(zobj,ele->ptr,reverse);
+    if (c->argc > 3 && !strcasecmp(c->argv[3]->ptr, "withscores")) {
```

Comment:
the new command format is `zrank key member [withscore]`, so `argc` should not be greater than 4, and if the last argument is not `withscore` we need reply the syntax error.